dvc/repo/experiments/executor/base.py

```diff
@@ -546,6 +535,17 @@ def reproduce(
         # multiprocessing calls
         return ExecutorResult(exp_hash, exp_ref, repro_force)
 
+    @staticmethod
+    def _repro_and_track(stage: "Stage", **kwargs) -> Optional["Stage"]:
+        from dvc.repo.reproduce import _reproduce_stage
+        from dvc.stage.utils import _get_stage_files
+
+        ret = _reproduce_stage(stage, **kwargs)
+        if not kwargs.get("dry") and (paths := _get_stage_files(stage)):
+            logger.debug("Staging stage-related files: %s", paths)
+            stage.repo.scm_context.add(paths)
+        return ret
```

Comment:
`stage.reproduce()` returns None for skipped (unchanged run-cached) stages, and when the entire pipeline is skipped `dvc.reproduce()` ends up returning an empty list. In `exp run` we still need to force git-tracking the `dvc.yaml` and `dvc.lock` for those skipped stages.